server/src/main/java/org/elasticsearch/action/get/TransportGetAction.java

```diff
@@ -298,8 +298,12 @@ private void tryGetFromTranslog(GetRequest request, IndexShard indexShard, Clust
     }
 
     static DiscoveryNode getCurrentNodeOfPrimary(ClusterState clusterState, ShardId shardId) {
```

Comment:
TBC, these missed corner cases were there from before, but the whole retrying logic and corner cases made it clear they were not handled properly.

Reply:
Yes :) But what I was referring to is that `getCurrentNodeOfPrimary` throwing or returning null if current node is not found (which is a bit inconsistent) was existing behaviour and not introduce by https://github.com/elastic/elasticsearch/pull/105003.